nucleus-sync/src/main/java/org/alfresco/hxi_connector/nucleus_sync/services/processors/UserMappingSyncProcessor.java

```diff
@@ -0,0 +1,160 @@
+                .collect(Collectors.toMap(AlfrescoUser::email, Function.identity()));
+
+        Map<String, IamUser> nucleusIamUserByEmail = nucleusIamUsers.stream()
+                .collect(Collectors.toMap(IamUser::email, Function.identity()));
+
+        Map<String, NucleusUserMappingOutput> nucleusMappingByAlfrescoId = currentUserMappings.stream()
+                .collect(
+                        Collectors.toMap(
+                                NucleusUserMappingOutput::externalUserId,
+                                Function.identity()));
+
+        List<String> nucleusMappingsToDelete = new ArrayList<>();
+        List<NucleusUserMappingInput> nucleusMappingsToCreate = new ArrayList<>();
+        List<UserMapping> cachedMappings = new ArrayList<>();
+        Set<String> validAlfrescoIds = new HashSet<>();
+
+        Set<String> allEmails = new HashSet<>();
+        allEmails.addAll(alfrescoUserByEmail.keySet());
+        allEmails.addAll(nucleusIamUserByEmail.keySet());
+
+        for (String email : allEmails)
+        {
+            AlfrescoUser alfrescoUser = alfrescoUserByEmail.get(email);
+            IamUser nucleusIamUser = nucleusIamUserByEmail.get(email);
+
+            boolean isAlfrescoUser = alfrescoUser != null;
+            boolean isNucleusUser = nucleusIamUser != null;
+
+            if (isAlfrescoUser && isNucleusUser)
+            {
+                String alfrescoUserId = alfrescoUser.id();
+                String nucleusUserId = nucleusIamUser.userId();
+
+                validAlfrescoIds.add(alfrescoUserId);
+                cachedMappings.add(
+                        new UserMapping(alfrescoUser.email(), alfrescoUserId, nucleusUserId));
+
+                if (!nucleusMappingByAlfrescoId.containsKey(alfrescoUserId))
+                {
+                    nucleusMappingsToCreate.add(
+                            new NucleusUserMappingInput(nucleusUserId, alfrescoUserId));
+                }
+            }
+        }
+
+        for (NucleusUserMappingOutput nucleusMapping : currentUserMappings)
+        {
+            String alfrescoUserId = nucleusMapping.externalUserId();
+            if (!validAlfrescoIds.contains(alfrescoUserId))
+            {
+                nucleusMappingsToDelete.add(alfrescoUserId);
+            }
+        }
+
+        executeUserBatchOperations(nucleusMappingsToDelete, nucleusMappingsToCreate);
+
+        if (LOGGER.isDebugEnabled())
+        {
+            LOGGER.debug("Final user mappings count: {}", cachedMappings.size());
+        }
+        return cachedMappings;
+    }
+
+    private void executeUserBatchOperations(
+            List<String> nucleusMappingsToDelete,
+            List<NucleusUserMappingInput> nucleusMappingsToCreate)
+    {
+        for (String alfrescoUserId : nucleusMappingsToDelete)
+        {
+            nucleusClient.deleteUserMapping(alfrescoUserId);
+        }
+        if (LOGGER.isDebugEnabled())
```

Comment:
This is PMD being very defensive. There's no need to do this when the method call just loads the size of a list. You could also satify PMD by using fluent logging:
https://www.slf4j.org/manual.html#fluent